### PR TITLE
Do not trim `options.originalText`

### PR DESCRIPTION
**Pull request by fisker**
Trim `originalText` doesn't make sense to me.

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-markdown/utils.js [hunk 1]
```diff
@@ -193,7 +193,7 @@ function hasGitDiffFriendlyOrderedList(node, options) {
 // workaround for https://github.com/remarkjs/remark/issues/351
 // leading and trailing newlines are stripped by remark
 function getFencedCodeBlockValue(node, originalText) {
-  const text = originalText.slice(
+  const text = originalText.trimEnd().slice(
```
Comment (fisker) [inline src/language-markdown/utils.js:196]: Finally, I understand why we still need this workaround.

Comment (fisker) [reply src/language-markdown/utils.js]: Fixing this in https://github.com/prettier/prettier/pull/8786

**Comment by thorn0** [comment]
Is this superseded by #8786?

**Comment by fisker** [comment]
Ah, closed by accident, I prefer merge this first, #8786 has some spec issue need discuss.

**Pull request merged**